packages/core/src/signals/Vector2CompoundSignalContext.ts

```diff
@@ -0,0 +1,247 @@
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  sub(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  dot(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  cross(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+  mod(
+    possibleVector: PossibleVector2,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ): Vector2 | TOwner | SignalGenerator<PossibleVector2, Vector2>;
+};
+
+export class Vector2CompoundSignalContext<
+  TOwner = void,
+> extends CompoundSignalContext<PossibleVector2, Vector2, 'x' | 'y', TOwner> {
+  public constructor(
+    entries: ('x' | 'y' | [keyof Vector2, Signal<any, any, TOwner>])[],
+    parser: (value: PossibleVector2) => Vector2,
+    initial: SignalValue<PossibleVector2>,
+    interpolation: InterpolationFunction<Vector2>,
+    owner: TOwner = <TOwner>(<unknown>undefined),
+    extensions: Partial<SignalExtensions<PossibleVector2, Vector2>> = {},
+  ) {
+    super(entries, parser, initial, interpolation, owner, extensions);
+
+    Object.defineProperty(this.invokable, 'edit', {
+      value: this.edit.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'scale', {
+      value: this.scale.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'mul', {
+      value: this.mul.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'div', {
+      value: this.div.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'add', {
+      value: this.add.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'sub', {
+      value: this.sub.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'dot', {
+      value: this.dot.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'cross', {
+      value: this.cross.bind(this),
+    });
+
+    Object.defineProperty(this.invokable, 'mod', {
+      value: this.mod.bind(this),
+    });
+  }
+
+  public override toSignal(): Vector2Signal<TOwner> {
+    return this.invokable;
+  }
+
+  public edit(
+    callback: (current: Vector2) => SignalValue<PossibleVector2>,
+    duration?: number,
+    timingFunction?: TimingFunction,
+    interpolationFunction?: InterpolationFunction<Vector2>,
+  ) {
+    const newValue = callback(this.invokable());
```

Comment:
Could just use `get`:
```suggestion
    const newValue = callback(this.get());
```